### DISCUSSION TITLE
Feature Request: extend the Github Content API to do Multi file operations in a single commit. from isaacs/github#199

### DISCUSSION THREAD

**Discussion by tx0c** [community/community#88796] [Apps, API and Webhooks]
### Select Topic Area

Product Feedback

### Body

want to CRUD multiple files in a single commit, and therefore in a single HTTP request with the [contents API](https://developer.github.com/v3/repos/contents/).

different solutions in #83252 was suggesting to tarball (tar or zip) of multiple files and send as 1 content over current contents API, but it would show just 1 tarball instead, losing all features of Github to show all raw code in its own file, all programming languages code syntax highlighting and functions references etc. all good features,
that doesn't solve the intention want to show multiple files tree in the repo;

https://docs.github.com/en/rest/gists/gists?apiVersion=2022-11-28#create-a-gist  something like this 1 API call to create gist with multiple files content;  can Github contents API be extended to upload multiple files in a single call?

https://github.com/isaacs/github/issues/199#issuecomment-531507388

some different thread suggested the Git Data API to commit changes, but that way would cause too many rounds of API interactions, eventually triggered API rate limit abuse...
>  to (potentially) hundreds of files in a Git repo. After about 100 requests they start getting rejected and flagged as abuse. Many of these 100's of files are just simple txt files. Is there a way of making a single request to create blobs for _many_ files?

> I've even tried throttling the # of requests so that it only submits 5 requests per second (for example). It still ends up rejecting them at some point for suspected abuse. I would have to throttle it to an unreasonable amount (taking 10 minutes or more) to get around this.

_Originally posted by @seanmcilvenna in https://github.com/isaacs/github/issues/199#issuecomment-531507388_

**Comment by ghostinhershell** [comment]
Thanks for sharing and contributing to our community @tx0c!

I’ll go ahead and file this with the applicable team so that it’s on their radar. In the meantime, you can check out our [public roadmap](https://github.com/github/roadmap/projects/1) to see the kinds of work we are focusing on for the coming year. Smaller fixes won’t be listed, but are often included when work is happening adjacent to it.

Thanks again!
